h2/src/main/org/h2/util/ValueHashMap.java

```diff
@@ -174,6 +191,42 @@ public V get(Value key) {
         }
         return list;
     }
+    
+    public Iterable<Map.Entry<Value, V>> entries() {
+        return new EntryIterable();
+    }
+    
+    private final class EntryIterable implements Iterable<Map.Entry<Value, V>> {
+        @Override
+        public Iterator<Map.Entry<Value, V>> iterator() {
+            return new EntryIterator();
+        }
+    }
+
+    private final class EntryIterator implements Iterator<Map.Entry<Value, V>> {
```

Comment:
An explicit no-arg constructor with default visibility in needed here for the same reason.